app/src/main/java/com/example/ncov19traking/ui/CovidCustomCard.kt

```diff
@@ -0,0 +1,86 @@
+package com.example.ncov19traking.ui
+
+import android.content.Context
+import android.graphics.Color
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.widget.TextView
+import androidx.core.content.ContextCompat
+import com.example.ncov19traking.R
+import com.google.android.material.card.MaterialCardView
+
+const val DEFAULT_TITLE_TEXT = "Title"
+const val DEFAULT_NUMBER_TEXT = "0"
+const val DEFAULT_PERCENTAGE_TEXT = "0%"
+const val CARD_ELEVATION = 10F
+const val CARD_RADIUS = 16F
+
+class CovidCustomCard @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : MaterialCardView(context, attrs, defStyleAttr) {
+
+    private var titleText = DEFAULT_TITLE_TEXT
+    private var numberText = DEFAULT_NUMBER_TEXT
+    private var percentageText = DEFAULT_PERCENTAGE_TEXT
+    private var numberTextColor = resources.getColor(R.color.recoveredColor)
+    private var percentageTextColor = resources.getColor(R.color.textColor)
+    private lateinit var titleTextView : TextView
+    private lateinit var numbersTextView : TextView
+    private lateinit var percentageTextView : TextView
+
+    init {
+        setupCard()
```

Comment:
so you are setting up view configuration before the inflation here? but at the same time you are waiting for inflation for other things on the `onFinishInflate` which one is it?.